server.js

```diff
@@ -1,23 +1,179 @@
 
-const welcomeMessage = {
-  id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
-};
+app.listen(process.env.PORT || PORT, function () {
+  console.log(`Server is listening on port ${PORT}. Ready to accept requests!`);
+});
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = require("./messages.json");
+const { response } = require("express");
 
+//  get the index file
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+// get /READ  all the messages
+app.get("/messages", function (request, response) {
+  response.status(200).json({ messages });
+});
+
+//  get / read a messsage by a specific ID
+
+app.get("/messages/:id", function (request, response) {
+  const found = messages.filter(
+    (eachmessage) => eachmessage.id === parseInt(request.params.id)
+  );
+
+  console.log(found);
+
+  if (found) {
+    response
+      .status(200)
+      .json(
+        messages.filter(
+          (eachmessage) => eachmessage.id === parseInt(request.params.id)
+        )
+      );
+  } else {
+    response.status(400).json({
+      message: `No message with the id of ${request.params.albumId}`,
+    });
+  }
+});
+
+//  - [ ] Create a new message /POST
+app.post("/CreateNewMessages", function (request, response) {
+  const createNewMessage = {
+    from: request.body.from,
+    text: request.body.text,
+  };
+  createNewMessage.id = messages.length 
+  console.log(createNewMessage);
+  if (!createNewMessage.from || !createNewMessage.text) {
+    return response.status(404).json({
+      message: `Please include a message ID, from who the message is sent & the text`,
+    });
+  } else {
+    messages.push(createNewMessage);
+    response.status(200).json(messages);
+  }
+});
+
+// - [ ] Delete a message, by ID
+
+app.delete("/messages/:id", function (request, response) {
+  const found = messages.filter(
+    (eachmessage) => eachmessage.id === parseInt(request.params.id)
+  );
+
+  if (found) {
+    response.status(200).json({
+      message: `DELETED  the message with the id of ${request.params.id}`,
+      messages: messages.filter(
+        (eachmessage) => eachmessage.id !== parseInt(request.params.id)
+      ),
+    });
+  } else {
+    response.status(400).json({
+      message: `No message with the id of ${request.params.albumId}`,
+    });
+  }
+});
+
+// chalalnge level 03
+
+// - [ ] Read _only_ messages whose text contains a given substring: `/messages/search?text=express`
+
+app.get("/search", (request, response) => {
+  // console.log(request.params.text)
+  // response.send(messages)
+
+  console.log(request.query);
+  const word = request.query.word.toLowerCase();
+  // console.log(word);
+
+  const filteredWord = messages.filter((eachmessage) =>
+    eachmessage.text.toLowerCase().includes(word)
+  );
```

Comment:
nice job adding a search route & checking for case too :) tweaking the route to be messages/search would be a good idea